pkg/apis/app/v1beta1/types.go

```diff
@@ -41,20 +42,21 @@ type FlinkApplicationSpec struct {
 	EntryClass         string                       `json:"entryClass,omitempty"`
 	ProgramArgs        string                       `json:"programArgs,omitempty"`
 	// Deprecated: use SavepointPath instead
-	SavepointInfo         SavepointInfo       `json:"savepointInfo,omitempty"`
-	SavepointPath         string              `json:"savepointPath,omitempty"`
-	DeploymentMode        DeploymentMode      `json:"deploymentMode,omitempty"`
-	RPCPort               *int32              `json:"rpcPort,omitempty"`
-	BlobPort              *int32              `json:"blobPort,omitempty"`
-	QueryPort             *int32              `json:"queryPort,omitempty"`
-	UIPort                *int32              `json:"uiPort,omitempty"`
-	MetricsQueryPort      *int32              `json:"metricsQueryPort,omitempty"`
-	Volumes               []apiv1.Volume      `json:"volumes,omitempty"`
-	VolumeMounts          []apiv1.VolumeMount `json:"volumeMounts,omitempty"`
-	RestartNonce          string              `json:"restartNonce"`
-	DeleteMode            DeleteMode          `json:"deleteMode,omitempty"`
-	AllowNonRestoredState bool                `json:"allowNonRestoredState,omitempty"`
-	ForceRollback         bool                `json:"forceRollback"`
+	SavepointInfo         SavepointInfo             `json:"savepointInfo,omitempty"`
+	SavepointPath         string                    `json:"savepointPath,omitempty"`
+	DeploymentMode        DeploymentMode            `json:"deploymentMode,omitempty"`
+	RPCPort               *int32                    `json:"rpcPort,omitempty"`
+	BlobPort              *int32                    `json:"blobPort,omitempty"`
+	QueryPort             *int32                    `json:"queryPort,omitempty"`
+	UIPort                *int32                    `json:"uiPort,omitempty"`
+	MetricsQueryPort      *int32                    `json:"metricsQueryPort,omitempty"`
+	Volumes               []apiv1.Volume            `json:"volumes,omitempty"`
+	VolumeMounts          []apiv1.VolumeMount       `json:"volumeMounts,omitempty"`
+	RestartNonce          string                    `json:"restartNonce"`
+	DeleteMode            DeleteMode                `json:"deleteMode,omitempty"`
+	AllowNonRestoredState bool                      `json:"allowNonRestoredState,omitempty"`
+	ForceRollback         bool                      `json:"forceRollback"`
+	MaxCheckpointRestoreAgeSec time.Duration        `json:"maxCheckpointRestoreAgeSec,omitempty"`
```

Comment:
I'm not sure using the time.Duration type will do the right thing. You might need to make this an int32 and do the conversion manually.